app/controllers/api/v1/instances_controller.rb

```diff
@@ -1,15 +1,9 @@
 # frozen_string_literal: true
 
-class Api::V1::InstancesController < Api::BaseController
-  skip_before_action :require_authenticated_user!, unless: :limited_federation_mode?
-  skip_around_action :set_locale
+class Api::V1::InstancesController < Api::V2::InstancesController
+  include DeprecationConcern
 
-  vary_by ''
-
-  # Override `current_user` to avoid reading session cookies unless in limited federation mode
-  def current_user
-    super if limited_federation_mode?
-  end
```

Comment:
Why were the removed lines moved to the v2 controller?

Reply:
An alternative would be to move have that logic be handled in `show` instead of in a class method. This could make sense as well if we have more specific deprecations to convey.

Reply:
Thanks for clarifying. I missed the changed inheritance. Now it makes sense, though it seems a bit unusual.

However, given it is deprecated anyway, I would not spend much more time (or thought) on this.